Funcky.Test/Monads/ResultTest.cs

```diff
@@ -131,6 +131,14 @@ public void SelectManyOnlyCallsSelectorWhenOk()
             FunctionalAssert.IsError(length);
         }
 
+        [Fact]
+        public void SelectManyWithOkResultMatchesTherightValue()
+            => FunctionalAssert.IsOk(2, Result.Ok(1).SelectMany(i => Result.Ok(i + 1)));
+
+        [Fact]
+        public void SelectManyWithErrorResultMatchesTherightValue()
```

Comment:
There's no test for `SelectMany` when it returns an `Error`

Reply:
Added here: 072d9eaa56477325a184fe68dbfdf945da71c4a3